### PR TITLE
Implement enough of geckolib to cascade the first node

### PR DESCRIPTION
**Pull request by bholley**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10697)

### CODE REVIEW

**Comment by bholley** [comment]
Added some additional stuff from today.

**Comment by bholley** [comment]
r? @SimonSapin

Renaming to reflect the fact that this is a catch-all for the geckolib work I did this week.

With this PR, we make it through the first cascaded node and panic in Servo_GetComputedValuesForAnonymousBox. @emilio is going to work on that next.

CC @heycam

**Comment by bholley** [comment]
@bors-servo retry

On phone.

**Comment by SimonSapin** [comment]
Style crate changes look good.

---

Reviewed 3 of 5 files at r8, 1 of 3 files at r10, 1 of 2 files at r14, 1 of 2 files at r20.
Review status: 6 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10697#-:-KFxEcRLQ_v_amg3VT7x:-1660219074)*

File: ports/geckolib/properties.mako.rs [hunk 1]
```diff
@@ -118,16 +118,35 @@ pub struct ${style_struct.gecko_struct_name};
...
 </%def>
 
+<%!
+def is_border_style_masked(ffi_name):
+    return ffi_name.split("[")[0] in ["mBorderStyle", "mOutlineStyle", "mTextDecorationStyle"]
+
+def get_gecko_property(ffi_name):
+    if is_border_style_masked(ffi_name):
+        return "(self.gecko.%s & (gecko_style_structs::BORDER_STYLE_MASK as u8))" % ffi_name
+    else:
```
Comment (emilio) [inline ports/geckolib/properties.mako.rs]: nit: unneeded return?

File: ports/geckolib/properties.mako.rs [hunk 2]
```diff
@@ -118,16 +118,35 @@ pub struct ${style_struct.gecko_struct_name};
...
+        return "self.gecko.%s" % ffi_name
+
+def set_gecko_property(ffi_name, expr):
+    if is_border_style_masked(ffi_name):
+        return "self.gecko.%s &= !(gecko_style_structs::BORDER_STYLE_MASK as u8);" % ffi_name + \
+               "self.gecko.%s |= %s as u8;" % (ffi_name, expr)
+    else:
```
Comment (emilio) [inline ports/geckolib/properties.mako.rs]: ditto

**Comment by bholley** [comment]
@emilio the return statements are necessary, because it's python code, and a regular def rather than a mako def. I used a regular def here to do inline expressions, since mako defs insert newlines before and after.

**Pull request merged**